packages/@lwc/babel-plugin-component/src/post-process/transform.js

```diff
@@ -35,7 +35,14 @@ module.exports = function postProcess({ types: t }) {
 
     function collectObservedFields(body, decoratedProperties) {
         const mappers = {
-            ObjectExpression: ({ properties }) => properties.map(({ key: { name } }) => name),
+            ObjectExpression: ({ properties }) =>
+                properties.map(({ key }) => {
+                    if (t.isIdentifier(key)) {
+                        return key.name;
+                    } else if (t.isStringLiteral(key)) {
```

Comment:
can we add a comment explaining how this works here.

Reply:
Check how ['a'] on the class body is processed

Reply:
they are added as a property. added a test and another check to filter out those cases on the observed fields.

Reply:
Why aren't to properties observed?